src/vector/backends/numpy.py

```diff
@@ -70,6 +70,15 @@ def _array_from_columns(columns: typing.Dict[str, ArrayLike]) -> ArrayLike:
         >>> _array_from_columns({"x": [1, 2, 3], "y": [1, 2, 4]})
         array([(1., 1.), (2., 2.), (3., 4.)], dtype=[('x', '<f8'), ('y', '<f8')])
     """
+
+    for coordinates in columns.values():
+        if not all(
+            isinstance(x, (int, float, numpy.float64, numpy.int64)) for x in coordinates
+        ):
+            raise TypeError(
+                "A coordinate must be of the type int, float, numpy.int64, or numpy.float64"
+            )
```

Comment:
Same thing as above, but for NumPy arrays instead of Awkward Arrays this time.

For NumPy arrays, you'll want to accept all the integer and floating point types:

```python
>>> dt = numpy.array([1, 2, 3]).dtype
>>> dt
dtype('int64')
>>> issubclass(dt.type, (numpy.integer, numpy.floating))
True
```

We want to exclude booleans, complex numbers, and datetime/timedeltas, as well as more exotic types like [structured arrays](https://numpy.org/doc/stable/user/basics.rec.html).

```python
>>> for t in [
...     numpy.bool_,
...     numpy.int8,
...     numpy.int16,
...     numpy.int32,
...     numpy.int64,
...     numpy.uint8,
...     numpy.uint16,
...     numpy.uint32,
...     numpy.uint64,
...     numpy.float16,   # not all platforms have this one
...     numpy.float32,
...     numpy.float64,
...     numpy.float128,   # not all platforms have this one
...     numpy.complex64,
...     numpy.complex128,
...     numpy.datetime64,
...     numpy.timedelta64,
... ]:
...     dt = numpy.dtype(t)
...     print(f"{dt} {issubclass(dt.type, (numpy.integer, numpy.floating))}")
... 
bool False
int8 True
int16 True
int32 True
int64 True
uint8 True
uint16 True
uint32 True
uint64 True
float16 True
float32 True
float64 True
float128 True
complex64 False
complex128 False
datetime64 False
timedelta64 True
```

Reply:
Also, Python's `int` and `float` are not possible types for data from a NumPy array. If you're trying to capture all inputs, start by casting the input as a NumPy array and do this test on the dtype (so that the array is converted only once; this is the same advice as for Awkward Arrays).

Incidentally, if you want to check to see if something in Python is an integer or floating-point number, use `numbers.Real` (or `numbers.Integral` for integers-only).

```python
>>> isinstance(5, numbers.Real)
True
>>> isinstance(3.14, numbers.Real)
True
```

Not all "numbers" are subclasses of Python's concrete built-in types. The `numbers` module has abstract types for anything number-like. For instance,

```python
>>> isinstance(numpy.float32(3.14), float)
False
>>> isinstance(numpy.float32(3.14), numbers.Real)
True
```